thirdparty/glib/CMakeLists.txt

```diff
@@ -3,65 +3,38 @@ list(APPEND PATCH_FILES
     glib-libglib-only.patch
     # Android makes Linux syscalls hard to get to... Fix pilfered from Python.
     glib-2.58.2-android-linux-syscalls.patch
+    # So we can install development files only.
+    fix-install-tags.patch
 )
-
-list(APPEND PATCH_CMD COMMAND autoreconf -vif)
-
-# Autotools needs some hints when x-compiling
-if(CHOST)
-    list(APPEND gen_cache_config glib_cv_stack_grows=no)
-    list(APPEND gen_cache_config glib_cv_uscore=no)
-    # In order to support legacy PocketBook 4.x firmware we should get rid of eventfd@GLIBC_2.7 and pipe2@GLIBC_2.9
-    if(LEGACY OR POCKETBOOK)
-        list(APPEND gen_cache_config glib_cv_eventfd=no)
-        list(APPEND gen_cache_config ac_cv_func_pipe2=no)
-        # As well as mkostemp@GLIBC_2.7
-        list(APPEND gen_cache_config ac_cv_func_mkostemp=no)
-    endif()
-    if(ANDROID)
-        # disable getpwuid, getgrgid and timegm for > android-12
-        list(APPEND gen_cache_config ac_cv_func_posix_getpwuid_r=no)
-        list(APPEND gen_cache_config ac_cv_func_posix_getgrgid_r=no)
-        list(APPEND gen_cache_config ac_cv_func_getpwuid_r=no)
-        list(APPEND gen_cache_config ac_cv_func_getgrgid_r=no)
-        list(APPEND gen_cache_config ac_cv_func_timegm=no)
-    endif()
```

Comment:
That part confused me: the comment about android-12 is from 23985d3, in 2016, so I assume 12 is for the API level? Android 3.1  (HONEYCOMB_MR1, 2011), and not Android 12 (S[now Cone], level 31, 2021). The flags themselves come from 23985d3 and 773db3f66 (via db68822f9 & e579d0eba).

Reply:
Yup, that's API level 12.

https://developer.android.com/guide/topics/manifest/uses-sdk-element.html#ApiLevels

Reply:
On that subject, this page has IMHO a better overview: https://apilevels.com/.